Keas.Mvc/Handlers/ApiKeyMiddleware.cs

```diff
@@ -45,16 +45,18 @@ public async Task Invoke(HttpContext context, ApplicationDbContext dbContext, IO
             }
 
             // lookup team that this API key has acess to
-            var teamAccess = dbContext.Teams.FirstOrDefault(t => t.ApiCode == headerGuidValue);
-
-            if (teamAccess == null)
+            var teamApi = dbContext.TeamApiCodes.Include(a => a.Team).FirstOrDefault(a => a.ApiCode == headerGuidValue);
+            if (teamApi == null || teamApi.Team == null)
             {
                 // no team found with your auth token, fail
                 context.Response.StatusCode = 401; //UnAuthorized
                 await context.Response.WriteAsync(InvalidApiKeyMessage);
                 return;
             }
 
+            var teamAccess = teamApi.Team;
```

Comment:
I think you could do this whole things with just replacing the first line with `var teamAccess = dbContext.Teams.FirstOrDefault(t => t. TeamApiCode.ApiCode == headerGuidValue);`